### ISSUE TITLE
[jvm-packages] XGBoost on Spark crashes with SIGSEV

### ISSUE THREAD

**Issue by marrrcin** [dmlc/xgboost#3935]
**This issue was mentioned here https://discuss.xgboost.ai/t/xgboost-on-spark-crashes-with-sigsev/475 and is suspected to be a bug.**

I use Scala in Azure Databricks with the following setup:

 - 5x worker node (`28.0 GB Memory, 8 Cores, 1.5 DBU`)
 - 1x driver (`14.0 GB Memory, 4 Cores, 0.75 DBU`)
 - runtime: `5.0 ML Beta (includes Apache Spark 2.4.0, Scala 2.11)`

I have a Spark Dataframe with 760k rows with two columns:

1. label (`Double`)
2. features (`SparseVector ` of length `84224` each)

I want to use `XGBoost` on my Dataframe, to train the regression model:

    val params = Map(
      "objective" -> "reg:linear",
      "max_depth" -> 6,
      "eval_metric" -> "rmse"
    )
    var model = new XGBoostRegressor(params)
      .setFeaturesCol("features")
      .setLabelCol("label")
      .setTreeMethod("approx")
      .setNumRound(20)
      .setNumEarlyStoppingRounds(3)
      .setUseExternalMemory(true)
      .setMaxDepth(6)
      .setNumWorkers(10)

    val trainedModel = model.fit(trainSample)

After launching, I get the following error:

> SIGSEGV (0xb) at pc=0x00007f62a9d33e0e, pid=3954,
> tid=0x00007f62c88db700

**What I've tried so far:**

When I set `numWorkers` to `1`, the training starts, but obviously runs really slow, which I believe is no the way it should be used.

The documentation here: https://xgboost.readthedocs.io/en/latest/tutorials/external_memory.html and here: https://docs.databricks.com/spark/latest/mllib/third-party-libraries.html#xgboost does not help at all with my case.

**My questions are:**

 1. Is it possible to run XGBoost on Dataset that is bigger than memory of each individual worker? (I assume that it's YES, but correct me if I'm wrong)
 1. How to use External Memory properly, so that when I take even bigger dataset XGBoost will do the training?
 1. Is partitioning of the input Dataframe impacting the training somehow?

**Comment by CodingCat** [comment]
Don’t use Spark 2.4 with XGB, it should not be able to run.....if you can run, that means databricks has customized xgb then I have no way to debug it

**Comment by CodingCat** [comment]
Please trybto reproduce the problem in a clean environment with spark 2.3

**Comment by CodingCat** [comment]
With xgboost 0.81

**Comment by marrrcin** [comment]
I've installed XGBoost from Spark Package (this method: https://docs.azuredatabricks.net/spark/latest/mllib/third-party-libraries.html#install-xgboost-spark-package) and only the following configuration worked for me:
XGBoost 0.8 with Spark 2.3.1
Other configurations:
* XGB 0.8.1 on Spark 2.3.1
* XGB 0.8.1 on Spark 2.3.0
* XGB 0.8 on Spark 2.3.0
Resulted in `error: object spark is not a member of package ml.dmlc.xgboost4j.scala`
